src/walletSdk/Anchor/Types.ts

```diff
@@ -0,0 +1,86 @@
+import { MemoType } from "stellar-sdk";
+import { TransactionStatus } from "../Watcher/Types";
+
+export interface AnchorServiceInfo {
+  deposit: { [key: string]: AnchorServiceAsset };
+  withdraw: { [key: string]: AnchorServiceAsset };
+  fee: { enabled: boolean };
+  features: { account_creation: boolean; claimable_balances: boolean };
+}
+
+interface BaseTransaction {
+  id: string;
```

Comment:
yup good catch https://github.com/stellar/typescript-wallet-sdk/pull/20/commits/5abb4d4caad8e89ccd51954a961bc1fb3664b17a

@Ifropc it looks it's also missing from the kotlin sdk (if I'm reading it right)

Reply:
Yep kind would be nice

Reply:
Oh I think I remembered, kind is only in error transaction, because all other transactions are already typed (it's either Deposit/Withdrawal/IncompleteDeposit/IncompleteWithdrawal)
The reason for incomplete is because they are missing a lot of fields, that are mandatory for any in-progress status transaction.
Not sure if it's user-friendly for typescript. 
In Kotlin, I can write
```kotlin
val tx: AnchorTransaction
when (tx) -> 
 is Deposit -> // handle deposit
 is IncompleteDeposit -> // handle incompelte deposit
 is Error -> // handle error
 else -> Exception("Unexpected kind") 
```